framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -50,6 +52,8 @@ MaterialPropertyInterface::MaterialPropertyInterface(const MooseObject * moose_o
     _mi_tid(_mi_params.get<THREAD_ID>("_tid")),
     _stateful_allowed(true),
     _get_material_property_called(false),
+    _block_restricted(block_ids.find(Moose::ANY_BLOCK_ID) == block_ids.end()),
```

Comment:
Yeah, looking through this PR, I don't see where these Booleans are used in this class. Only the derived class where you have access to the other objects for this information.